File: hello_xr/platform/platformplugin_posix.cpp
#include "platformplugin.h"
#include "common.h"

// Create a platform plugin for the platform specified at compile time.
void XR_PLATFORM_init(struct Options* options, struct PlatformData* data) { Log::Write(Log::Level::Info, "PLATFORM => POSIX"); }
void XR_PLATFORM_deinit() {}

// OpenXR instance-level extensions required by this platform.
const char** XR_PLATFORM_GetInstanceExtensions(size_t* n) {
    *n = 0;
    return nullptr;
}

// Perform required steps after updating Options
void XR_PLATFORM_UpdateOptions(struct Options* options) {}

// Provide extension to XrInstanceCreateInfo for xrCreateInstance.
XrBaseInStructure* XR_PLATFORM_GetInstanceCreateExtension() { return nullptr; }

File: hello_xr/platform/platformplugin_posix.zig
const c = @import("c");

pub fn init() void {
    // Log::Write(Log::Level::Info, "PLATFORM => POSIX");
}

pub fn deinit() void {}

// OpenXR instance-level extensions required by this platform.
pub fn GetInstanceExtensions() [][*:0]const u8 {
    return &.{};
}

// Provide extension to XrInstanceCreateInfo for xrCreateInstance.
pub fn GetInstanceCreateExtension() ?*c.XrBaseInStructure {
    return null;
}
